Index: configure
===================================================================
--- configure	(revision 26087)
+++ configure	(working copy)
@@ -2401,6 +2401,7 @@
     os/2*)
         strip="lxlite"
         ln_s="cp -f"
+        objformat="aout"
         add_ldflags -Zomf -Zbin-files -Zargs-wild -Zmap
         SHFLAGS='$(SUBDIR)$(NAME).def -Zdll -Zomf'
         FFSERVERLDFLAGS=""
Index: libavcodec/x86/fft_mmx.asm
===================================================================
--- libavcodec/x86/fft_mmx.asm	(revision 26087)
+++ libavcodec/x86/fft_mmx.asm	(working copy)
@@ -71,7 +71,7 @@
     %1
 %endmacro
 
-section .text align=16
+SECTION_TEXT
 
 %macro T2_3DN 4 ; z0, z1, mem0, mem1
     mova     %1, %3
Index: libavcodec/x86/dsputil_yasm.asm
===================================================================
--- libavcodec/x86/dsputil_yasm.asm	(revision 26087)
+++ libavcodec/x86/dsputil_yasm.asm	(working copy)
@@ -28,7 +28,7 @@
 pb_zzzz3333zzzzbbbb: db -1,-1,-1,-1,3,3,3,3,-1,-1,-1,-1,11,11,11,11
 pb_zz11zz55zz99zzdd: db -1,-1,1,1,-1,-1,5,5,-1,-1,9,9,-1,-1,13,13
 
-section .text align=16
+SECTION_TEXT
 
 %macro PSWAPD_SSE 2
     pshufw %1, %2, 0x4e
Index: libavcodec/x86/x86inc.asm
===================================================================
--- libavcodec/x86/x86inc.asm	(revision 26087)
+++ libavcodec/x86/x86inc.asm	(working copy)
@@ -62,11 +62,22 @@
     %elifidn __OUTPUT_FORMAT__,macho
         SECTION .text align=%1
         fakegot:
+    %elifidn __OUTPUT_FORMAT__,aout
+        section .text
     %else
         SECTION .rodata align=%1
     %endif
 %endmacro
 
+; aout does not support align=
+%macro SECTION_TEXT 0-1 16
+    %ifidn __OUTPUT_FORMAT__,aout
+        SECTION .text
+    %else
+        SECTION .text align=%1
+    %endif
+%endmacro
+
 %ifdef WIN64
     %define PIC
 %elifndef ARCH_X86_64
